### ISSUE TITLE
[sig-network] Proxy version v1 A set of valid responses are returned for both pod and service ProxyWithPath [Conformance]

### ISSUE THREAD

**Issue by pacoxu** [kubernetes/kubernetes#107374]
##### Error text:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:630
Jan  4 07:05:01.070: Pod didn't start within time out period
Unexpected error:
    <*errors.errorString | 0xc0001ca2b0>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/proxy.go:314
```
#### Recent failures:
[2022/1/6 上午1:25:16 ci-kubernetes-kind-conformance-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-kind-conformance-parallel/1478778946777518080)
[2022/1/6 上午12:43:32 ci-kubernetes-e2e-windows-containerd-gce-20h2](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-containerd-gce-20h2/1478769130575237120)
[2022/1/5 下午11:03:53 ci-kubernetes-e2e-windows-gce-20h2-1-23](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce-20h2-1-23/1478744049920774144)
[2022/1/5 下午7:14:51 ci-kubernetes-e2e-windows-gce-2019](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce-2019/1478686417042280448)
[2022/1/5 下午3:14:51 ci-kubernetes-e2e-windows-gce-2019](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce-2019/1478626018292928512)

/kind failing-test

/sig network

**Comment by fedebongio** [comment]
@thockin curious why you think is API Machinery and not Network, can you explain please? Thanks!

**Comment by fedebongio** [comment]
Ping @thockin (sorry!)

**Comment by aojea** [comment]
> Ping @thockin (sorry!)

@fedebongio  it test the proxy feature of the apiserver (too many proxies :smile: ) 

> urlString := f.ClientConfig().Host + "/api/v1/namespaces/" + ns + "/pods/agnhost/proxy/some/path/with/" + httpVerb

**Comment by aojea** [comment]
I can take it 
/assign

**Comment by fedebongio** [comment]
Thank you @aojea !!
/triage accepted

**Comment by eddiezane** [comment]
This is still flaking quite a bit.

https://storage.googleapis.com/k8s-triage/index.html?test=%5Bsig-network%5D%20v1%20A%20set%20of%20valid%20responses%20are%20returned%20for%20both%20pod%20and%20service%20ProxyWithPath

**Comment by pacoxu** [comment]
dup with #109037 which was closed by a fix #109039
- Use WaitForPodRunningInNamespace for flakey proxy e2e tests. It will increase the time out from [1 * time.Minute](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/proxy.go#L61) to [5 * time.Minute](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/pod/wait.go#L53), which is expected to deflake the tests.

The below failure was fixed. See https://storage.googleapis.com/k8s-triage/index.html?test=%5Bsig-network%5D%20v1%20A%20set%20of%20valid%20responses%20are%20returned%20for%20both%20pod%20and%20service%20ProxyWithPath#363fdcde164a98bbdaad
```
82 test failures (0 today)
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:185
Mar 18 02:29:00.922: Unexpected error:
    <*errors.errorString | 0xc0002cc2b0>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:243
```

However, below is not fixed.  See https://storage.googleapis.com/k8s-triage/index.html?test=%5Bsig-network%5D%20v1%20A%20set%20of%20valid%20responses%20are%20returned%20for%20both%20pod%20and%20service%20ProxyWithPath#1c2b34000c29e8ba47f6
```
/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:630
Mar 24 18:44:37.987: Pod didn't start within time out period
Unexpected error:
    <*errors.errorString | 0xc000242280>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
occurred
/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/proxy.go:314
```

**Comment by pacoxu** [comment]
After some log investigation, some failures on the windows node(I have no experience with that.) and some failed for configmap mount.

Take https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-stable1-latest-upgrade-cluster-parallel/1509562493612068864 for example.

```
E0331 16:48:59.330503    1698 projected.go:286] Couldn't get configMap proxy-8750/kube-root-ca.crt: configmap "kube-root-ca.crt" not found
E0331 16:48:59.330521    1698 projected.go:192] Error preparing data for projected volume kube-api-access-lbbb6 for pod proxy-8750/agnhost: configmap "kube-root-ca.crt" not found
E0331 16:48:59.330584    1698 nestedpendingoperations.go:335] Operation for "{volumeName:kubernetes.io/projected/1de7ce0a-bc2b-40c3-955f-9b6c9707bd9d-kube-api-access-lbbb6 podName:1de7ce0a-bc2b-40c3-955f-9b6c9707bd9d nodeName:}" failed. No retries permitted until 2022-03-31 16:48:59.830563011 +0000 UTC m=+332.361355407 (durationBeforeRetry 500ms). Error: MountVolume.SetUp failed for volume "kube-api-access-lbbb6" (UniqueName: "kubernetes.io/projected/1de7ce0a-bc2b-40c3-955f-9b6c9707bd9d-kube-api-access-lbbb6") pod "agnhost" (UID: "1de7ce0a-bc2b-40c3-955f-9b6c9707bd9d") : configmap "kube-root-ca.crt" not found
I0331 16:48:59.330618    1698 event.go:294] "Event occurred" object="proxy-8750/agnhost" fieldPath="" kind="Pod" apiVersion="v1" type="Warning" reason="FailedMount" message="MountVolume.SetUp failed for volume \"kube-api-access-lbbb6\" : configmap \"kube-root-ca.crt\" not found"

```
